src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -165,60 +164,14 @@ def add_latest_version(
         else:
             try:
                 if (
-                    (latest_version := latest_value.get("version"))
-                    and (item_version := item.get("version"))
+                    (latest_version := latest_value.get("purl").version)  # type: ignore[union-attr]
+                    and (item_version := item.get("purl").version)  # type: ignore[union-attr]
```

Comment:
Instead of ignoring the types, can you add a check to make sure `latest_value.get("purl")` and  `item.get("purl")` are not `None`?

Reply:
I'm not sure I can. All of these fail to make mypy happy:
```
if (latest_value.get("purl") is not None and item.get("purl") is not None):
    if (
        (latest_version := latest_value.get("purl").version)
        and (item_version := item.get("purl").version)
        and version.Version(latest_version) < version.Version(item_version)
    ):
        latest_deps[key] = item
```
        
```
if (
    latest_value.get("purl") is not None
    and item.get("purl") is not None
    and (latest_version := latest_value.get("purl").version)
    and (item_version := item.get("purl").version)
    and version.Version(latest_version) < version.Version(item_version)
):
    latest_deps[key] = item
```

```
if latest_value.get("purl") is not None and item.get("purl") is not None:
    latest_value_purl: PackageURL = latest_value.get("purl")
    item_purl: PackageURL = item.get("purl")
    if (
        (latest_version := latest_value_purl.version)
        and (item_version := item_purl.version)
        and version.Version(latest_version) < version.Version(item_version)
    ):
        latest_deps[key] = item
```

Reply:
It seems this works:
```
latest_value_purl = latest_value.get("purl")
item_purl = item.get("purl")
    if (
        latest_value_purl is not None
        and item_purl is not None
        and (latest_version := latest_value_purl.version)
        and (item_version := item_purl.version)
        and version.Version(latest_version) < version.Version(item_version)
    ):
        latest_deps[key] = item
```